frontend/www/js/omegaup/course/student.ts

```diff
@@ -1,26 +1,27 @@
 import course_ViewStudent from '../components/course/ViewStudent.vue';
-import { OmegaUp } from '../omegaup';
+import { omegaup, OmegaUp } from '../omegaup';
+import { types } from '../api_types';
 import * as api from '../api';
 import * as UI from '../ui';
 import T from '../lang';
 import Vue from 'vue';
 
 OmegaUp.on('ready', function() {
-  var payload = JSON.parse(document.getElementById('payload').innerText);
+  const payload = types.payloadParsers.StudentsProgressPayload();
 
-  var initialStudent = null;
-  if (payload.students && payload.students.length > 0) {
-    initialStudent = payload.students[0];
-    for (var student of payload.students) {
+  let initialStudent: types.CourseStudent | null = null;
+  if (payload.students && Object.values(payload.students).length > 0) {
+    initialStudent = Object.values(payload.students)[0];
+    for (let student of Object.values(payload.students)) {
```

Comment:
```suggestion
    for (const student of Object.values(payload.students)) {
```

en general preferimos usar `const` para todo, a menos que sea necesario reasignar la variable. (curiosamente, dentro de un `for` no se considera reasignación por alguna razón).

Reply:
Done